test/testuninitvar.cpp

```diff
@@ -3079,6 +3079,35 @@ class TestUninitVar : public TestFixture {
                        "    } catch(...) {}\n"
                        "}");
         ASSERT_EQUALS("[test.cpp:4]: (error) Uninitialized variable: i\n", errout.str());
+
+        checkUninitVar("void f(bool x) {\n"
+                       "    bool b;\n"
+                       "    {\n"
+                       "        auto g = []{};\n"
+                       "        b = x;\n"
+                       "    }\n"
+                       "    if (b) {}\n"
+                       "}\n");
+        ASSERT_EQUALS("", errout.str());
+
+        checkUninitVar("void f(bool x) {\n"
+                       "    bool b;\n"
+                       "    {\n"
+                       "        int i[2]{ 1, 2 };\n"
+                       "        b = x;\n"
+                       "    }\n"
+                       "    if (b) {}\n"
+                       "}\n");
+        ASSERT_EQUALS("", errout.str());
+
+        checkUninitVar("void f(bool x) {\n"
+                       "    bool b;\n"
+                       "    {\n"
+                       "        auto g = []{};\n"
+                       "    }\n"
+                       "    if (b) {}\n"
+                       "}\n");
+        ASSERT_EQUALS("[test.cpp:6]: (error) Uninitialized variable: b\n", errout.str());
```

Comment:
Is this covered by valueFlowUninit?

Reply:
Let me check that for you...
No, it is not. The warning comes from `checkIfForWhileHead()`-

Reply:
We should probably open an issue to track that case.

Reply:
Actually `valueFlowUninit` does diagnose this, although its not entirely clear.